src/utils.ts

```diff
@@ -134,9 +43,28 @@ export function getRequestId(options: QueryOptions, target: QueryTarget) {
   return `${options.dashboardId}_${options.panelId}_${target.refId}`;
 }
 
-export function reduceToMap(array: { property: string; value: any }[]) {
-  return array.reduce((obj, { property, value }) => {
-    obj[property] = value;
-    return obj;
-  }, {});
-}
+export const applyFilters = <T>(objs: T[], filters: ParsedFilter[]): T[] => {
+  if (!filters.length) {
+    return objs;
+  }
+
+  return objs.filter(obj => filters.every(filter => checkFilter(obj, filter)));
+};
+
+export const checkFilter = <T>(obj: T, { path, filter, value }: ParsedFilter): boolean => {
+  const valueToFilter = get(obj, path, null);
+  const regex = new RegExp(`^${value}$`, 'i');
+
+  if (valueToFilter === null) {
+    return false;
+  }
+
+  switch (filter) {
+    case FilterType.RegexEquals:
+      return valueToFilter.match(regex);
+    case FilterType.RegexNotEquals:
+      return !valueToFilter.match(regex);
+    case FilterType.NotEquals:
+      return value.toLowerCase() !== valueToFilter.toLowerCase();
```

Comment:
I agree, but I guess user might not think so.
Doing this, I was thinking about this issue https://github.com/cognitedata/cognite-grafana-datasource/issues/65
Do we really need "strict not equal" for `!=`  ?

Reply:
https://github.com/cognitedata/cognite-grafana-datasource/issues/65
for this we probably just need to allow flags
because if we change it to case insensitive now it might break some dashboards..

Reply:
got it, so I'll redo `!=` to be more strict